pkg/operator/rules.go

```diff
@@ -24,10 +24,28 @@ import (
 	"github.com/pkg/errors"
 	"github.com/prometheus/prometheus/model/rulefmt"
 
+	"github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring"
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/informers"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+	namespacelabeler "github.com/prometheus-operator/prometheus-operator/pkg/namespace-labeler"
 	thanostypes "github.com/thanos-io/thanos/pkg/store/storepb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
 )
 
+type PromRuleCommonFields struct {
```

Comment:
Tried this approach to avoid passing all these fields to function. Also looking for better name for this struct :)

Reply:
I'd suggest to leave the initialization of the label selector and namespace labeler to the caller. It will make the struct more lightweight + Prometheus & Thanos don't create the same ns labeler (because of the partial response strategy).

```golang
type PrometheusRuleSelector struct {
   labelSelector labels.Selector
   nsLabeler namespacelabeler.Labeler
   rulesInfs *informers.ForResource
   logger log.Logger
   metrics *Metrics
}

func NewPrometheusRuleSelector(...) *PrometheusRuleSelector {
...
}

func (prs *PrometheusRuleSelector) Select(namespaces ...string) (map[string]string, error) {
...
}
```